site/docs/uk/hosting/vps.md

```diff
@@ -52,130 +152,631 @@ systemd --- це потужний менеджер служб, який пере
+
+::: tip Вебсервер
+Ми використовуємо Caddy, тому що, на відміну від поширених вебсерверів, як-от Nginx або Apache, він автоматично налаштовує SSL-сертифікати.
+Це значно спрощує посібник.
+Втім, ви можете вибрати будь-який вебсервер, який вам подобається.
+:::
+
+#### Встановлення
+
+Наступні 5 команд завантажать й автоматично запустять Caddy як службу systemd з назвою `caddy`.
 
 ```sh
-systemctl stop <назва-служби>
+apt install -y debian-keyring debian-archive-keyring apt-transport-https curl
+curl -1sLf 'https://dl.cloudsmith.io/public/caddy/stable/gpg.key' | gpg --dearmor -o /usr/share/keyrings/caddy-stable-archive-keyring.gpg
+curl -1sLf 'https://dl.cloudsmith.io/public/caddy/stable/debian.deb.txt' | tee /etc/apt/sources.list.d/caddy-stable.list
+apt update
+apt install caddy
 ```
 
-#### Вимкнення запуску при завантаженні
+> Дивіться [посібник зі встановлення Caddy](https://caddyserver.com/docs/install) для інших варіантів встановлення.
+
+Перевірте статус роботи Caddy:
 
 ```sh
-systemctl disable <назва-служби>
+systemctl status caddy
 ```
 
-## PM2 (тільки для Node.js)
+:::details Усунення помилок
+Деякі хостинг-провайдери надають VPS із передвстановленим вебсервером, наприклад, [Apache](https://httpd.apache.org/).
+Кілька вебсерверів не можуть одночасно працювати на одній машині.
+Щоб Caddy працював, потрібно зупинити й вимкнути інший вебсервер:
 
-PM2 --- це менеджер демонів процесів для Node.js, який допоможе вам керувати та тримати ваш застосунок у режимі онлайн 24/7.
+```sh
+systemctl stop apache2
+systemctl disable apache2
+```
+
+:::
+
+Тепер, якщо ви відкриєте у браузері IP-адресу свого сервера, то побачите типову сторінку з інструкцією з налаштуванню Caddy.
+
+#### Конфігурування
+
+Щоб Caddy обробляв запити, які надходять до нашого домену, потрібно змінити конфігурацію Caddy.
 
-### Встановлення
+Виконайте наступну команду, щоб відкрити файл конфігурації Caddy:
 
 ```sh
-npm install pm2@latest -g
+nano /etc/caddy/Caddyfile
+```
 
-# Якщо ви використовуєте Yarn
-yarn global add pm2
+Ви побачите наступну типову конфігурацію:
+
+```text
+# The Caddyfile is an easy way to configure your Caddy web server.
+#
+# Unless the file starts with a global options block, the first
+# uncommented line is always the address of your site.
+#
+# To use your own domain name (with automatic HTTPS), first make
+# sure your domain's A/AAAA DNS records are properly pointed to
+# this machine's public IP, then replace ":80" below with your
+# domain name.
+
+:80 {
+  # Set this path to your site's directory.
+  root * /usr/share/caddy
+
+  # Enable the static file server.
+  file_server
+
+  # Another common task is to set up a reverse proxy:
+  # reverse_proxy localhost:8080
+
+  # Or serve a PHP site through php-fpm:
+  # php_fastcgi localhost:9000
+}
+
+# Refer to the Caddy docs for more information:
+# https://caddyserver.com/docs/caddyfile
+```
+
+Для роботи бота приведіть конфіг до наступного вигляду:
```

Comment:
```suggestion
Для обробки вхідних запитів на домен, треба додати наступний код у кінець файла:
```

Reply:
```suggestion
Для обробки вхідних запитів, приведіть конфіг до наступного вигляду:
```

Reply:
I agree. It is up to @deptyped to make the final decision.